fashion-messaging-app-api/server.js

```diff
@@ -37,38 +37,47 @@ app.get('/photos', async (req, res) => {
   
   try {
     const response = await clientAPI.photos.search({ query, per_page: PHOTOS });
-   
-    var myHeaders = new Headers();
-    myHeaders.append("x-api-key", "4d552e9f30522b1dec7c712f83c67c235be86e25ec14b4ba3493383ec7b81d3f");                               
-
-    let fetchPromises = await Promise.all(response.photos.map(async photo => { 
-      var formdata = new FormData();
-      formdata.append("image_url", photo.src.original);
-    
-
-      var requestOptions = {
-        method: 'POST',
-        headers: myHeaders,
-        body: formdata,
-      };
-
-      try {
-        const fetchResponse = await fetch('https://cloudapi.lykdat.com/v1/detection/items', requestOptions);
-        const result = await fetchResponse.json();     
+    let fetchPromises = response.photos.map(async photo => {
+      if (cache.has(photo.src.original)) {
+        return cache.get(photo.src.original);
```

Comment:
Wondering if you really need to store the whole photo object as your value in the hashtable. You may be able to have a much more succinct value